src/pharmpy/tools/amd/run.py

```diff
@@ -964,38 +1003,165 @@ def _run_allometry(model):
 
 
 def _subfunc_iov(amd_start_model, occasion, strictness, path) -> SubFunc:
-    if occasion is None:
-        warnings.warn('IOVsearch will be skipped because occasion is None.')
-        return noop_subfunc
+    def _run_iov(model):
+        res = run_tool(
+            'iovsearch',
+            model=model,
+            results=model.modelfit_results,
+            column=occasion,
+            strictness=strictness,
+            path=path / 'iovsearch',
+        )
+        assert isinstance(res, Results)
+        return res
+
+    return _run_iov
 
-    if occasion not in amd_start_model.dataset:
+
+def validate_input(
+    model: Model,
+    results: Optional[ModelfitResults] = None,
+    modeltype: str = 'basic_pk',
+    administration: str = 'oral',
+    strategy: Optional[str] = "SIRIAC",
+    cl_init: float = 0.01,
+    vc_init: float = 1.0,
+    mat_init: float = 0.1,
+    b_init: Optional[Union[int, float]] = None,
+    emax_init: Optional[Union[int, float]] = None,
+    ec50_init: Optional[Union[int, float]] = None,
+    met_init: Optional[Union[int, float]] = None,
+    search_space: Optional[str] = None,
+    lloq_method: Optional[str] = None,
+    lloq_limit: Optional[str] = None,
+    allometric_variable: Optional[Union[str, sympy.Symbol]] = None,
+    occasion: Optional[str] = None,
+    path: Optional[Union[str, Path]] = None,
+    resume: bool = False,
+    strictness: Optional[str] = "minimization_successful or (rounding_errors and sigdigs>=0.1)",
+    retries_strategy: Literal["final", "all_final", "skip"] = "final",
+):
+    to_be_skipped = []
+
+    if not (isinstance(results, ModelfitResults) or results is None):
+        raise ValueError(f"Results need to be of type 'ModelfitResults. Not {type(results)}")
+
+    if modeltype not in ALLOWED_MODELTYPE:
+        raise ValueError(f'Invalid input: "{modeltype}" as modeltype is not supported')
+
+    if administration not in ALLOWED_ADMINISTRATION:
         raise ValueError(
-            f'Invalid `occasion`: got `{occasion}`,'
-            f' must be one of {sorted(amd_start_model.datainfo.names)}.'
+            f"Administration need to be one of {ALLOWED_ADMINISTRATION}. Not '{administration}'"
         )
 
-    def _run_iov(model):
-        if occasion not in model.dataset:
-            warnings.warn(f'Skipping IOVsearch because dataset is missing column "{occasion}".')
-            return None
+    if strategy not in ALLOWED_STRATEGY:
+        raise ValueError(f"Strategy need to be one of {ALLOWED_STRATEGY}. Not '{strategy}'")
+
+    for param_init in [
+        cl_init,
+        vc_init,
+        mat_init,
+        b_init,
+        emax_init,
+        ec50_init,
+        met_init,
+    ]:
+        if not (isinstance(param_init, (float, int)) or param_init is None):
```

Comment:
Shouldn't it be possible here to use: https://github.com/pharmpy/pharmpy/blob/d32c15c9fc8b8abc2e7d5a93e51078dac18b8b10/src/pharmpy/internals/fn/type.py#L20